src/app/core/services/draft/draft-repository.service.ts

```diff
@@ -73,13 +73,14 @@ export class DraftRepositoryService {
     );
   }
 
+  /**
+   * @returns true if draft does not contain any data
+   */
   async isDraftEmpty(draft: RegistrationDraft) {
-    if (hasAnyObservations(draft)) {
-      return false;
-    }
-
-    const attachments = await firstValueFrom(this.newAttachmentSerivice.getAttachments(draft.uuid));
-    return attachments.length === 0;
+    const anyObservations = hasAnyObservations(draft);
+    const anyExistingAttatchments = draft.registration.Attachments?.length > 0;
+    const anyNewAttachments = (await firstValueFrom(this.newAttachmentSerivice.getAttachments(draft.uuid)))?.length > 0;
+    return !anyObservations && !anyExistingAttatchments && !anyNewAttachments;
```

Comment:
Er det ikke fint å returnere false så tidlig som mulig hvis man kan? Jeg syns det er fint å gjøre funksjonen et par linjer lenger og returnere false etter de to første sjekkene også, så slipper vi å sjekke filsystemet hver gang?

Reply:
Veldig bra innspill, Jørgen. Ytelse er viktig! Jeg fikser